test/functional/lua/ui_spec.lua

```diff
@@ -67,5 +69,15 @@ describe('vim.ui', function()
       eq('Inputted text', result[1])
       eq('Input: ', result[2])
     end)
+
+    it('does not cause error on nil opt', function()
+      feed(':lua vim.ui.input(nil, function() end)<cr>')
+      eq('', eval('v:errmsg'))
```

Comment:
This could be a false negative so let's go a bit further and actually feed input into the prompt, then check the result.

Like this (not tested):

```suggestion
      feed(':lua test_result=vim.ui.input(nil, function() end)<cr>')
      feed('x')
      eq('x', exec_lua('return test_result'))
```

Reply:
`vim.ui.input` doesn't return anything, so I made the callback set the variable.